src/zkevm_specs/evm/execution/returndatacopy.py

```diff
@@ -0,0 +1,47 @@
+from ..instruction import Instruction, Transition
+from ..table import CallContextFieldTag, RW, CopyDataTypeTag
+
+
+def returndatacopy(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+
+    memory_offset_word, offset_word, size_word = (
+        instruction.stack_pop(),
+        instruction.stack_pop(),
+        instruction.stack_pop(),
+    )
+
+    return_data_length = instruction.call_context_lookup(
+        CallContextFieldTag.LastCalleeReturnDataLength, RW.Read
+    )
+    return_data_offset = instruction.call_context_lookup(
+        CallContextFieldTag.LastCalleeReturnDataOffset, RW.Read
+    )
+
+    memory_offset, size = instruction.memory_offset_and_length(memory_offset_word, size_word)
+    next_memory_size, memory_expansion_gas_cost = instruction.memory_expansion_dynamic_length(
+        memory_offset, size
+    )
+    gas_cost = instruction.memory_copier_gas_cost(size, memory_expansion_gas_cost)
+
+    copy_rwc_inc, _ = instruction.copy_lookup(
+        instruction.curr.call_id,
```

Comment:
It should copy from callee's memory instead of itself, and that's the main reason `LastCalleeId` is also stored in call context.

```suggestion
        instruction.call_context_lookup(CallContextFieldTag.LastCalleeId),
```

Reply:
Hi, Han! Thanks for review! I chose call_id because I thought the data was already copied by opcode `return` to current call's memory space, correct me if I am wrong.

Reply:
Not really, so `RETURN` doesn't actually copy data, it's `*CALL*` that could be specified to copy return data to certain memory chunk (e.g. https://github.com/ethereum/go-ethereum/blob/master/core/vm/instructions.go#L699-L701).

However, the return data is still immutable and accessible until next call/create, and that's the actual source of `RETURNDATACOPY` (the [`interpreter.returnData`](https://github.com/ethereum/go-ethereum/blob/master/core/vm/instructions.go#L340)).

Reply:
Got it. will fix soon! Also will fix retundatacopy circuit accordingly later.